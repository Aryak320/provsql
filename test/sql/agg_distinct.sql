\set ECHO none
SET search_path TO public,provsql;

CREATE TABLE agg_result AS
  SELECT count(distinct classification)
    FROM (SELECT name,classification FROM personnel ORDER BY name) b;

SELECT remove_provenance('agg_result');

SELECT * FROM agg_result;

SELECT aggregation_formula(count,'personnel_name') FROM agg_result;

CREATE TABLE agg_result2 AS
  SELECT city, count(distinct classification)
    FROM personnel
    GROUP BY city ORDER BY city, classification;

SELECT remove_provenance('agg_result2');

SELECT * FROM agg_result2 ORDER BY city;

SELECT city, aggregation_formula(count,'personnel_name') FROM agg_result2 ORDER BY city;

DROP TABLE agg_result;
DROP TABLE agg_result2;
