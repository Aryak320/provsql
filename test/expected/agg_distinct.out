\set ECHO none
remove_provenance

(1 row)
city|count
Berlin|1 (*)
New York|2 (*)
Paris|3 (*)
(3 rows)
city|aggregation_formula
Berlin|sum{ (1 * δ((Ellen ⊕ Susan))) }
New York|sum{ (1 * δ(John)) , (1 * δ(Paul)) }
Paris|sum{ (1 * δ(Nancy)) , (1 * δ(Dave)) , (1 * δ(Magdalen)) }
(3 rows)
